vspace/src/main/java/edu/asu/diging/vspace/web/publicview/ExhibitionSlideController.java

```diff
@@ -79,21 +86,33 @@ public String slide(Model model, @PathVariable("slideId") String slideId, @PathV
         model.addAttribute("firstSlide", module.getStartSequence().getSlides().get(0).getId());
         String nextSlideId = "";
         String prevSlideId = "";
-        int slideIndex = 0;
-        slideIndex = sequenceSlides.indexOf(sildeManager.getSlide(slideId));
+        ISlide currentSlide = sildeManager.getSlide(slideId);
+        int slideIndex = sequenceSlides.indexOf(currentSlide);
         int slideSize = sequenceSlides.size();
         if (slideSize > slideIndex + 1) {
             nextSlideId = sequenceSlides.get(slideIndex + 1).getId();
         }
         if (slideIndex > 0) {
             prevSlideId = sequenceSlides.get(slideIndex - 1).getId();
         }
+        model.addAttribute("sequences",moduleManager.getModuleSequences(moduleId));
         model.addAttribute("sequence",sequenceExist);
         model.addAttribute("slides", sequenceSlides);
         model.addAttribute("currentSequenceId", sequenceId);
         model.addAttribute("nextSlide", nextSlideId);
         model.addAttribute("prevSlide", prevSlideId);
-        model.addAttribute("currentSlideCon", sildeManager.getSlide(slideId));
+        model.addAttribute("currentSlideCon", currentSlide);
+        if(currentSlide instanceof BranchingPoint) {
+            model.addAttribute("choices", ((BranchingPoint)currentSlide).getChoices());
+        }
+        if(back) {
+            choiceHistory.removeLastElementFromSequenceSlideHistory();
+        }
+        if(choiceHistory.getFromSequenceSlideHistory().size()>0) {
+            model.addAttribute("showBackToPreviousChoice", true);
+            model.addAttribute("previousChoiceId", choiceHistory.peekTopChoiceId());
```

Comment:
this is the previous squence id, right? not the choice id?

Reply:
The user will be navigating to the next sequence from choice(choiceID is nothing but ID linked to sequenceID). previouschoice made more sense to me as we say we are linking choices to branching points and using those to navigate to the linked sequences from branching points. your thoughts?

Reply:
previous choice id sounds like you're storing the id of the click branch point choice, but that's not what you're storing right? the id will literally hold the id of the previous sequence, right? so that's what I would call that.